### PR TITLE
Use async workspace methods over sync ones.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
WIP.  Do not review yet.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Test vsi please

**Comment by CyrusNajmabadi** [comment]
Test vsi please

**Comment by CyrusNajmabadi** [comment]
retest prtest/lin/dbg/unit32 please

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by davkean** [comment]
https://github.com/dotnet/roslyn/issues/6981 is the cause of the VSI failures.

retest vsi please

**Comment by CyrusNajmabadi** [comment]
retest prtest/win/dbg/eta  please

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide Please review this.

**Comment by Pilchie** [comment]
Also tagging @dotnet/roslyn-interactive, but it LGTM.  Worth testing VSI?

**Comment by CyrusNajmabadi** [comment]
test vsi please

**Comment by amcasey** [comment]
Seems like @rchande should take a look.

File: src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractLanguageService`3.IVsImmediateStatementCompletion2.cs [hunk 1]
```diff
@@ -68,7 +71,36 @@ int IVsImmediateStatementCompletion2.InstallStatementCompletion(int install, IVs
...
+                        {
+                            this.filters[textView].RemoveContext();
+                        }
+                    }
+                });
+
+            return VSConstants.S_OK;
+        }
+
+        private async Task<ValueTuple<IVsTextLines, AbstractDebuggerIntelliSenseContext>> GetCompletionContextAsync(
```
Comment (rchande) [inline src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractLanguageService`3.IVsImmediateStatementCompletion2.cs]: Does this need to return a tuple? The caller doesn't use the returned IVsTextLines.

File: src/VisualStudio/Core/Def/ServicesVSResources.resx [hunk 2]
```diff
@@ -513,4 +513,7 @@ Use the dropdown to view and switch to other projects this file may belong to.</
   <data name="SynchronizingWithClassView" xml:space="preserve">
     <value>Synchronizing with {0}...</value>
   </data>
+  <data name="Loading_immediate_window" xml:space="preserve">
+    <value>Loading immediate window</value>
```
Comment (rchande) [inline src/VisualStudio/Core/Def/ServicesVSResources.resx]: Trivial: This code runs to initialize intellisense in the "watch" window too--maybe pick a more generic phrase like "Loading IntelliSense" or a second phrase for the watch window?

**Comment by rchande** [comment]
LGTM. Can you make sure the *DebuggerIntelliSense VSI tests still pass?

**Comment by CyrusNajmabadi** [comment]
test vsi please

File: src/VisualStudio/Core/Def/ServicesVSResources.resx (hunk 2 above)
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/ServicesVSResources.resx]: Why_the_underscores?

**Comment by CyrusNajmabadi** [comment]
retest vsi please

**Comment by davkean** [comment]
prtest/win/vsi/p2 is caused by: https://github.com/dotnet/roslyn/issues/6863

**Comment by jasonmalinowski** [comment]
@CyrusNajmabadi, this was obsoleted by your other PR?

**Pull request closed without merging**